src/listeners/browser.js

```diff
@@ -1,16 +1,101 @@
+import eventsBulkRequest from '../services/events/bulk';
+import eventsService from '../services/events';
+import impressionsBulkRequest from '../services/impressions/bulk';
+import impressionsService from '../services/impressions';
+import { fromImpressionsCollector } from '../services/impressions/dto';
+import logFactory from '../utils/logger';
+
+const log = logFactory('splitio-client:cleanup');
+
+// 'unload' event is used instead of 'beforeunload', since 'unload' is not a cancelable event, so no other listeners can stop the event from occurring.
+const UNLOAD_DOM_EVENT = 'unload';
+
 /**
- * Not implemented yet.
+ * We'll listen for 'unload' event over the window object, since it's the standard way to listen page reload and close.
+ *
  */
 export default class BrowserSignalListener {
-  constructor() {
-    // noop
+
+  constructor(context) {
+    this.storage = context.get(context.constants.STORAGE);
+    this.settings = context.get(context.constants.SETTINGS);
+    this.flushData = this.flushData.bind(this);
   }
 
+  /**
+   * start method. 
+   * Called when SplitFactory is initialized. 
+   * We add a handler on unload events. The handler flushes remaining impressions and events to the backend.
+   */
   start() {
-    // noop
+    log.debug('Registering flush handler when unload page event is triggered.');
```

Comment:
Move the log inside the if, since it could be logged but not be true. That would be very confusing when we're debugging with a customer.

Consider adding a message with an else clause too